walnut/converter/dump.json

```diff
@@ -0,0 +1,364 @@
+[INFO] Using config file 'json:///home/alidock/.sw/slc8_x86-64/QualityControl/master-1/etc/basic.json'
+2020-06-04 16:59:04.933371     QC infologger initialized
+2020-06-04 16:59:04.933395     QC version 0.26.1
+2020-06-04 16:59:04.933554     policyName : tst-raw
+2020-06-04 16:59:04.933588     Configuration loaded : 
+2020-06-04 16:59:04.933595     >> Task name : QcTask
+2020-06-04 16:59:04.933599     >> Module name : QcSkeleton
+2020-06-04 16:59:04.933604     >> Detector name : TST
+2020-06-04 16:59:04.933614     >> Cycle duration seconds : 10
+2020-06-04 16:59:04.933619     >> Max number cycles : -1
+2020-06-04 16:59:04.933736     >> Check name : QcCheck
+2020-06-04 16:59:04.933843     >> Inputs (1): QC_QcTask-mo_0  checks (1): QcCheck  stores (1): QC_QcTask-mo_0 
+{
+    "workflow": [
+        {
+            "name": "producer-0",
+            "inputs": [],
+            "outputs": [
+                {
+                    "binding": "out",
+                    "origin": "TST",
+                    "description": "RAWDATA",
+                    "subspec": 0,
+                    "lifetime": 0
+                }
+            ],
+            "options": [],
+            "rank": 0,
+            "nSlots": 1,
+            "inputTimeSliceId": 0,
+            "maxInputTimeslices": 1
+        },
+        {
+            "name": "Dispatcher",
+            "inputs": [
```

Comment:
The correspondence is calculated by DPL, based on the actual InputSpec and OutputSpec. The channel names are indeed `from_{sender}_to_{receiver}-{extra}`. Bind vs connect or the channel type is define by a policy. For the latter two I guess you can do whatever you think it's best for your setup. For the channel name, I guess I will have to expose those?

Reply:
I'm not sure about the policy but from what I can guess:
- `bind` is only ever used for tasks
- `connect` is only used for workflows
- `bind` forms a `push` channel by default
- `connect` forms a `pull` channel by default

Let me know if these sound correct to you. 

Exposing the channel names would be great. I have not been able to infer what the `{extra}` field should be from the DPL Dumps.

Reply:
regarding bind / connect at the moment we have that sources always bind and sinks always connect, but that's just how it is setup by default.

Reply:
same for push / pull (actually now I use a pair, IIRC...) source always push, sink always pull, but again, in certain cases they might be customised.

Reply:
@ktf How would it be reflected in the DPL dump if they were customized?

Reply:
I now see that this part is not exposed, as it was meant to be populated by the Control. Maybe I should add the list of channels for you to consume?

Reply:
@ktf Yes please, this would be most useful. What we do is essentially building the `--channel-config`, but as a tree for `FairMQOptions` instead of as a command line param.

Reply:
Done in AliceO2Group/AliceO2#3870. Let me know if it helps.